### PR TITLE
Handle errors

### PR DESCRIPTION
**Pull request by ash2k**
Some missing error handling.

Edit: ok, GoLand found 391 unhandled errors in addition to these 😱 You can see for yourself by running Code -> Run inspection by name... and selecting "Unhandled error".

### CODE REVIEW

File: api/krusty/kustomizer.go [hunk 1]
```diff
@@ -90,19 +90,25 @@ func (b *Kustomizer) Run(
 		return nil, err
 	}
 	if b.options.DoLegacyResourceSort {
-		builtins.NewLegacyOrderTransformerPlugin().Transform(m)
+		err = builtins.NewLegacyOrderTransformerPlugin().Transform(m)
+		if err != nil {
+			return nil, err
+		}
```
Comment (monopole) [inline api/krusty/kustomizer.go:96]:
```
if err = builting.New...; err != nil {
```

File: api/krusty/kustomizer.go [hunk 2]
```diff
@@ -90,19 +90,25 @@ func (b *Kustomizer) Run(
...
+				konfig.ManagedbyLabelKey: fmt.Sprintf("kustomize-%s", provenance.GetProvenance().Semver()),
+			},
 			FieldSpecs: []types.FieldSpec{{
 				Path:               "metadata/labels",
 				CreateIfNotPresent: true,
 			}},
 		}
-		t.Transform(m)
+		err = t.Transform(m)
+		if err != nil {
```
Comment (monopole) [inline api/krusty/kustomizer.go:109]: ditto

**Review by monopole — COMMENTED** [review]
~3/4 of the unchecked error returns are in tests, for the most part _intentionally_ ignored because the `if err !=nil ` thing gets tedious and there are usually tests for error paths.  The places that ignore the error could be sprinkled with `assert.NoError(whatever)` if you'd like to spend some time on it.

In the non-test code goland's analysis sometimes is off (e.g. if an error is printed it's not considered handled`).  

Sometimes it's just a choice because the underlying code is now calling some kyaml function that has an error in its sig, but in the situation at hand the args cannot trigger an error (or the whole program would have died earlier) or the encapsulating call doesn't return an error or cannot fix it. 

And sometimes like these two you found it's easy to pass the error up and avoid the complaint.
n such cases, the options are to swallow (not good) or panic, which at least offers a clue as to the line number of the error.

**Comment by monopole** [comment]
hey, it's fine.  These are good, and if you want to continue, please do!
/lgtm

**Comment by ash2k** [comment]
@monopole Ok, cool. I will continue =) I want a solid library so it's in my own interest to fix the issues. I'll see if I can tweak the linter to make it catch issues like that.

**Pull request merged**